daliuge-engine/dlg/deploy/create_dlg_job.py

```diff
@@ -636,6 +636,15 @@ def main():
         default=None,
     )
 
+    parser.add_option(
+        "--ssh_key",
+        dest="ssh_key",
+        type="string", 
+        action="store",
+        help="Path to ssh private key",
+        default=""
+    )
+
```

Comment:
I'm sure you're basically basing yourself on the options above, but it seems like most of the explicitly-given parameters here are unnecessary (`dest`, `type`, `action` and `default` are all superfluous, with a `default`-less `add_option` call defaulting to `None`, which then again ties to my other comment). But maybe you want to be consistent across all options added to the parser, so feel free to keep it as it is if you think that's best.

Reply:
This is a great point, I shouldn't just be defaulting to the code above. I'll rectify this.

Reply:
Speaking of defaulting - I also agree the default should be `None`, so I'll keep that consistent with the `paramiko` API as per your other comment.